.buildkite/publish.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+set -euv
+
+# The build agents have gpg 2.0.22, which doesn't have the `--pinentry-mode` option, but
+# the sbt-pgp plugin assumes that 2.x has it, and generates an invalid command.
+# We can either create a wrapper script that removes that option, or update gpg
+# somewhere in pipeline.nix
+
+# Force a restart of the agent becuase it may be out of sync with what Nix installed.
+gpgconf --kill all
+
+gpg --version
+
+
+# The build agent might have this key from before.
+GPG_EXISTS=$(gpg --list-keys "$GPG_KEY_ID" >&2 && echo "yes" || echo "no")
+
+if [[ "$GPG_EXISTS" == "no" ]]; then
+	echo "$GPG_KEY" | base64 --decode | gpg --batch --import
+fi
+# Local testing showed that without this the SBT plugin got "Bad passphrase".
+gpg --passphrase $GPG_PASSPHRASE --batch --yes -a -b LICENSE
+
+# https://github.com/olafurpg/sbt-ci-release#secrets
+export PGP_SECRET="$GPG_KEY"
+export PGP_PASSPHRASE="$GPG_PASSPHRASE"
+export SONATYPE_USERNAME="$OSS_USERNAME"
+export SONATYPE_PASSWORD="$OSS_PASSWORD"
+
+set +u
+
+#https://github.com/sbt/sbt/issues/3570
+export JAVA_OPTS="$JAVA_OPTS -Dsbt.gigahorse=false"
+
+# ci-release cannot be called on individual modules, but it looks like
+# with `publish / skip := true` in build.sbt for the default project,
+# without any aggregation, by default it would publish nothing, so
+# let's tell it here by using `sbt-ci-release` env vars.
+# NOTE: +rlp/publishSigned with the `+` would cross publish,
+# but it doesn't work because of scapegoat (see below).
+export CI_SNAPSHOT_RELEASE="; bytes/publishSigned; rlp/publishSigned; crypto/publishSigned"
+export CI_RELEASE=$CI_SNAPSHOT_RELEASE
+export CI_SONATYPE_RELEASE=$"; bytes/sonatypeBundleRelease; rlp/sonatypeBundleRelease; crypto/sonatypeBundleRelease"
+
+# Scala 2.12 has up to scapegoat 1.4.5, while Scala 2.13 starts with 1.4.7.
+# I couldn't make build.sbt vary the scapegoat version by the current cross build,
+# so as a workaround the combos are called here explicitly.
+function release {
+    SCALA_VERSION=$1
+    export SCAPEGOAT_VERSION=$2
+
+    sbt "++ $SCALA_VERSION ; ci-release"
+}
+
+function releaseAll {
+    release 2.12.10 1.4.5
+    release 2.13.4 1.4.7
```

Comment:
oh this duplication with build.sbt will come back to bite us....

Reply:
It shouldn't be too hard to move more of this logic into sbt.

Reply:
You'd think so, right! I spent hours trying to figure out with a code like this:

```
scapegoatVersion := (scalaBinaryVersion.value match {
  case "2.12" => "1.4.5"
  case "2.13" => println("Still 2.13"); "1.4.7"
})
```
I start `sbt`, type `++ 2.12.10` and see "Still 2.13" printed every time, then I type `scalaBinaryVersion` and it says `2.12.10`, really frustrating.

Reply:
If you want to give it a go, you should be able to try wit `sbt +compile` for example (or at least `sbt +publishLocal`), the `+` should trigger cross building, and then it fails because it tries to grab the 1.4.7 version for 2.12.10, even when I moved the `scapegoatVersion` under `node`. 

If I removed `in ThisBuild` then it wants to grab 1.0.0. No idea what I'm missing, this [should work](https://stackoverflow.com/questions/41403305/in-sbt-how-can-i-cross-build-with-dependencies-that-are-not-required-in-one-ver).